lib/src/components/NxBreadcrumb/types.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { HTMLAttributes } from 'react';
+import * as PropTypes from 'prop-types';
+
+export interface Crumb {
+  name: string;
+  href: string;
+}
+
+export interface DropdownProps {
+  crumbs: Crumb[];
+  isOpen: boolean;
+  onToggleCollapse: () => void;
+}
+
+export interface StatefulProps extends HTMLAttributes<HTMLElement> {
+  crumbs: Crumb[];
+}
+
+export interface Props extends StatefulProps {
+  isDropdownOpen: boolean;
+  onToggleDropdown: () => void;
+}
+
+export const statefulPropTypes: PropTypes.ValidationMap<StatefulProps> = {
+  crumbs: PropTypes.arrayOf(PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    href: PropTypes.string.isRequired
+  }).isRequired).isRequired
```

Comment:
And for additional clarification, the first `.isRequired` on line 34 refers to the shape of each crumb, and the second `.isRequired` refers to the array of crumbs being a required prop?

Reply:
> And for additional clarification, the first .isRequired on line 34 refers to the shape of each crumb, and the second .isRequired refers to the array of crumbs being a required prop?

Yeah all propTypes validator accept `null` and `undefined` by default in addition to what they normally do, and you have to use the variation attached to their `isRequired` property to avoid that.  So here the inner `isRequired` effectively means that the array children cannot be `null` or `undefined`, and the outer one means that the array itself cannot be `null` or `undefined`

Reply:
excellent, thanks!